carbon-design-system/carbon-charts
# Enhance svelte chart wrappers, add TypeScript support

Closes #922 

### Updates
- export `chart`, `id`, `ref` props from `BaseChart`
- fix default type for `data` in `BaseChart` to be consistent with `ChartTabularData`
- bind `chart`, `ref` props in Svelte chart wrappers
- add missing `ComboChart`
- add TypeScript support
  - install `typescript`, `@tsconfig/svelte`, `svelte-check` as development dependencies
  - add `packages/svelte/types` folder to contain TypeScript definitions
  - add `packages/svelte/tests` folder to test TypeScript definitions
  - use `svelte-check` (official Svelte package) to validate TypeScript tests
  - update `package.json#peerDependencies` svelte version from `3.20.x` to `3.31.x` [1]

To validate the TypeScript definitions, run `yarn test` or `yarn svelte-check --workspace=tests` in `packages/svelte`. 0 errors and warnings should be emitted.

**Notes**

- [1] Svelte version >=v3.31.x is required to use the charts with TypeScript

### Demo screenshot or recording

Running the Svelte storybook locally:

<img width="1500" alt="Screen Shot 2021-01-12 at 3 44 22 PM" src="https://user-images.githubusercontent.com/10718366/104387999-2cc4a800-54ed-11eb-81a3-0534496cc361.png">


### Review checklist (for reviewers only)
- [ ] Demos all features
- [ ] Documented/annotated
- [ ] Matches UI/UX specs
- [ ] Meets the [code style guide](https://github.com/carbon-design-system/carbon-charts/wiki/Code-style-guide)
- [ ] Accessible
- [ ] Mobile first (responsive)
- [ ] RTL support (bidirectional text)
- [ ] Performant (limited bloat)
